clusterloader2/README.md

```diff
@@ -1,21 +1,79 @@
+go run cmd/clusterloader.go --kubeconfig=kubeConfig.yaml --testconfig=config.yaml
 ```
 Flags kubeconfig and testconfig are necessary.
 
-#### Flags
+### Flags
 
+#### Required
+
+These flags are required for any test to be run.
  - kubeconfig - path to the kubeconfig file.
- - testconfig - path to the test config file.
+ - testconfig - path to the test config file. This flag can be used multiple times
+if more than one test should be run.
 
-### Vendor
+#### Optional
 
-Vendor is created using [govendor].
+ - nodes - number of nodes in the cluster.
+If not provided, test will assign the number of available schedulable cluster nodes.
+ - report-dir - path to directory, where summaries files should be stored.
+If not specified, summaries are printed to standard log.
+
+## Tests
+
+### Test definition
+
+Test definition is an instantiation of this [api] (in json or yaml).
+The motivation and description of the API can be find in [design doc].
+Definitions of test as well as definitions of individual objects support templating.
+Templates for test definition come with one predefined value - ```{{.Nodes}}```, which represents the number of schedulable nodes in the cluster. \
+Example of a test definition can be found here: [load test].
+
+### Object template
 
+Object templates are similar to standard kubernetes object templates. The only difference is that the templating has been added.
+Parameters can be passed from the test definition to the object template using the ```templateFillMap``` map.
+Two always available parameters are ```{{.Name}}``` and ```{{.Index}}``` which specifies object name and object replica index respectively. \
+Example of a template can be found here: [load rc template].
+
+## Measurement
+
+Currently available measurements are:
+- **APIResponsiveness** \
+This metric creates summary for latency and number for server api calls. Api calls are divided by resource, subresource, verb and scope.
+- **CPUProfile** \
+This metric gathers the cpu usage profile provided by pprof for a given component.
+- **EtcdMetrics** \
+EtcdMetrics creates summary for etcd operations durations. These duration are represented as histograms.
+- **MemoryProfile** \
+This metric gathers the memory profile provided by pprof for a given component.
+- **MetricsForE2E** \
+The metric gathers e2e metrics from server, controller manager, all kubelets (optional based on metric parameter) and scheduler.
+- **SchedulingMetrics** \
+SchedulingMetrics summarizes the scheduler latency metrics for predicate evaluation, priority evaluation, preemption evaluation and binding.
+- **TestPhaseTimer** \
+TestPhaseTimer allows to create multiple instances of a timer for measuring time.
+Each timer is started and stopped using the measurement call defined the test definition. TestPhaseTimer results will be logged into a file.
+- **Timer** \
+Timer allows to measure time. The result will be written to standard log.
```

Comment:
I think I'm still a bit lost.
What is the difference between using Timer and TestPhaseTimer. What I should use each of them?

Reply:
- Timer measures time between "start" and "stop" calls. Result will only by printed to standard log. It's like an additional info for the log.
- TestPhaseTimer is one instance that create multiple timers. Basically each timer works in similar way to Timer measurement. The difference is that TestPhaseTimer can create one summary file with all measured times - it can be used for reporting the test phase execution times.

Reply:
So what's the point of having Timer at all?

I can use TestPhaseTimer and just call start and stop once, right?

Reply:
Timer was created as a first measurement. Currently it is redundant. I will remove it.

Reply:
OK - now it parses for me.

When you will be removing it, please also rename TestPhaseTimer to Timer then :)

Reply:
Changing TestPhaseTimer -> Timer: #239

Reply:
Timer (the old one) removed from README.